cmd/otel-allocator/config/config.go

```diff
@@ -39,10 +39,12 @@ const DefaultResyncTime = 5 * time.Minute
 const DefaultConfigFilePath string = "/conf/targetallocator.yaml"
 
 type Config struct {
-	LabelSelector      map[string]string  `yaml:"label_selector,omitempty"`
-	Config             *promconfig.Config `yaml:"config"`
-	AllocationStrategy *string            `yaml:"allocation_strategy,omitempty"`
-	FilterStrategy     *string            `yaml:"filter_strategy,omitempty"`
+	LabelSelector          map[string]string  `yaml:"label_selector,omitempty"`
+	Config                 *promconfig.Config `yaml:"config"`
+	AllocationStrategy     *string            `yaml:"allocation_strategy,omitempty"`
+	FilterStrategy         *string            `yaml:"filter_strategy,omitempty"`
+	PodMonitorSelector     *map[string]string `yaml:"pod_monitor_selector,omitempty"`
+	ServiceMonitorSelector *map[string]string `yaml:"service_monitor_selector,omitempty"`
```

Comment:
What's the benefit of making this a pointer to a map instead of `map[string]string`?

Reply:
I originally did this to make it so that these were optional, but that may not be necessary. Will rework :)